File: packages/shared-types/message.ts
import { z } from "zod";

export const removedItemSchema = z.object({
  kind: z.string(),
  value: z.string()
});

export const messageInputSchema = z.object({
  match_id: z.string().min(8).max(128),
  sender_pseudo_id: z.string().min(8).max(128),
  recipient_pseudo_id: z.string().min(8).max(128),
  raw_text: z.string().min(1).max(2000),
  match_status: z.enum(["pre", "post"]),
  post_match_contact_consent: z.boolean().default(false)
});

export const messageOutputSchema = z.object({
  scrubbed_text: z.string().default(""),
  removed: z.array(removedItemSchema).default([]),
  blocked: z.boolean(),
  reason: z.string().nullable().default(null)
});

export type RemovedItem = z.infer<typeof removedItemSchema>;
export type MessageInput = z.infer<typeof messageInputSchema>;
export type MessageOutput = z.infer<typeof messageOutputSchema>;

File: services/agent-service/src/embodied_agents/schemas/message.py
from typing import Literal

from pydantic import BaseModel, Field


class RemovedItem(BaseModel):
    kind: str
    value: str


class MessageInput(BaseModel):
    sender_pseudo_id: str = Field(min_length=8, max_length=128)
    recipient_pseudo_id: str = Field(min_length=8, max_length=128)
    raw_text: str = Field(min_length=1, max_length=2000)
    match_status: Literal["pre", "post"]
    post_match_contact_consent: bool = False


class MessageOutput(BaseModel):
    scrubbed_text: str = ""
    removed: list[RemovedItem] = Field(default_factory=list)
    blocked: bool
    reason: str | None = None
